sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
```

Comment:
`Datum.Next` on any type backed by `time.Time` would be invalid when `Datum.Next() < Datum`. Why not use that definition?